### PR TITLE
[IMP] mail: introduce ChatWindowHeaderView/onClick

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2831082

### CODE REVIEW

File: addons/mail/static/src/models/chat_window_header_view.js [hunk 1]
```diff
@@ -2,10 +2,32 @@
...
+                isEventHandled(ev, 'ChatWindowHeader.ClickShiftPrev') ||
+                isEventHandled(ev, 'ChatWindow.onClickHideMemberList') ||
+                isEventHandled(ev, 'ChatWindow.onClickShowMemberList')
+            ) {
+                return;
+            }
+            if (!this.chatWindowOwner.isVisible) {
+                this.chatWindowOwner.onClickFromChatWindowHiddenMenu();
+            } else {
+                this.chatWindowOwner.onClickHeader();
```
Comment (oomsveta) [inline addons/mail/static/src/models/chat_window_header_view.js:27]: Having a handler that calls another handler feels a bit odd to me. I'd recommend either moving the code of `onClickFromChatWindowHiddenMenu` and `onClickHeader` in here, or renaming it to something more declarative that would make the extra indirection worthwhile.

Comment (oomsveta) [reply addons/mail/static/src/models/chat_window_header_view.js:27]: How about the second suggestion?

Comment (alexkuhn) [reply addons/mail/static/src/models/chat_window_header_view.js:27]: Can you provide example? For me, these methods are specificity code for the handler, so keeping the `onX` naming convention still make sense.

**Review by oomsveta — APPROVED** [review]

**Comment by alexkuhn** [comment]
Moved to grouped PR: https://github.com/odoo/odoo/pull/91698

**Pull request closed without merging**